include/xtensor/xmath.hpp

```diff
@@ -2841,7 +2839,11 @@ namespace xt
             keep_dim_shape[el] = 1;
         }
         auto mrv = reshape_view<XTENSOR_DEFAULT_LAYOUT>(std::move(inner_mean), std::move(keep_dim_shape));
-        return nanmean<result_type>(square(cast<result_type>(sc) - std::move(mrv)), std::forward<X>(axes), es);
+        return nanmean<result_type>(
+            square(cast<result_type>(std::forward<E>(e)) - std::move(mrv)),
+            std::forward<X>(axes),
+            es
+        );
```

Comment:
This last bit breaks in a way I struggle understanding. The output of  `square` seems correct, but the output of `nanmean` is wrong.
Weirdly, removing the `es` solves the problem...